enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -353,6 +362,11 @@ On AWS STS, the `STSIAMRoleARN` would be something like:
 `arn:aws:iam::269733383066:role/newstscluster-openshift-logging-role-name`<br>
 This provides the role information linked by policy to access various cloud resources.
 
+### CCO Changes for HyperShift
+
+The kubeconfig of the worker node is mounted at a specied path on the CCO pod. 
+CCO can consume that kubeconfig when it starts the go client.
+
 ### Risks and Mitigations
```

Comment:
need to add some risks+mitigations in which we look at the resource consumption of the CCO, determine if it's acceptable to the hypershift/rosa team, and if not, potentially pursue how we can reduce the footprint.

Reply:
are you thinking comparing the utilization of "CCO can watch multiple clusters" vs "multiple CCOs each watching one cluster" ? or more broadly even comparing options beyond the scope of this EP like the pod identity webhook?

Reply:
no, i wasn't trying to go that far, just calling it out as:

1) we need to measure it
2) hypershift/rosa team needs some threshold defined that is acceptable/unacceptable
3) if we are exceeding that threshold, work will be required.  That work may be as simple as just going into the CCO code and finding optimizations we can make to reduce its footprint.  (possibly optimizations we can make that are specific to manual/STS mode).  If we find ourselves having to do more sophisticated things like a shared CCO across clusters, or switching back to using webhooks, that's going to significantly derail things.  So i guess you can also mention those options, but i do not expect us to be going down those paths.  (Shared CCO probably isn't even an option, from a tenancy/isolation requirements perspective)

Reply:
The CCO on build farms is taking up enormous amounts of memory footprint. It looks like the CCO keeps the following in cache:

- secrets
- configmaps
- namespaces

This is cluster-wide, always. We should be able to make the following optimizations:

- watch the two configmaps that need to be watched - the legacy cco config one and the one for leader-election (should be doable in c-r cache lib)
- do not cache namespaces, we just want to watch for creation to start filling in credentials (not sure if possible)
- label secrets created by the CCO, restrict all secret LIST+WATCH to that label selector (doable in c-r)

Reply:
https://github.com/openshift/cloud-credential-operator/pull/544 handles the configmap situation

Handling secrets will require a couple releases for labeling existing data and ratcheting validation to kick in - if we backport this, could be fast. LMK what we want.

Namespaces, I think we will need to just use a metadata-only watch if we want that functionality, but namespace objects are small so I'm not sure how much of a difference it will make

Reply:
https://github.com/openshift/cloud-credential-operator/pull/545 adds a controller to start labelling things